xonsh/environ.py

```diff
@@ -611,24 +614,29 @@ def load_static_config(ctx):
     return conf
 
 
-def xonshrc_context(rcfile=None, execer=None):
+def xonshrc_context(rcfiles=None, execer=None):
     """Attempts to read in xonshrc file, and return the contents."""
-    if rcfile is None or execer is None or not os.path.isfile(rcfile):
+    if (rcfiles is None or execer is None
+       or sum([os.path.isfile(rcfile) for rcfile in rcfiles]) == 0):
         return {}
-    with open(rcfile, 'r') as f:
-        rc = f.read()
-    if not rc.endswith('\n'):
-        rc += '\n'
-    fname = execer.filename
-    env = {}
-    try:
-        execer.filename = rcfile
-        execer.exec(rc, glbs=env)
-    except SyntaxError as err:
-        msg = 'syntax error in xonsh run control file {0!r}: {1!s}'
-        warn(msg.format(rcfile, err), RuntimeWarning)
-    finally:
-        execer.filename = fname
+    for rcfile in rcfiles:
+        try:
+            with open(rcfile, 'r') as f:
+                rc = f.read()
+            if not rc.endswith('\n'):
+                rc += '\n'
+            fname = execer.filename
+            env = {}
+            try:
+                execer.filename = rcfile
+                execer.exec(rc, glbs=env)
+            except SyntaxError as err:
+                msg = 'syntax error in xonsh run control file {0!r}: {1!s}'
+                warn(msg.format(rcfile, err), RuntimeWarning)
+            finally:
+                execer.filename = fname
+        except:
```

Comment:
Why is this outer try-except block here?  It seems like this loop should check `if not os.path.isfile(rcfile): continue` at the top. Also, there shouldn't be bare exceptions, flake8 complains :)